File: generate_test_images.py
from PIL import Image, ImageDraw
import os

# 生成拼图碎片图片
def generate_puzzle_piece():
    # 创建一个100x100的红色图片
    image = Image.new('RGBA', (100, 100), (255, 0, 0, 255))
    draw = ImageDraw.Draw(image)
    
    # 添加一些细节使其看起来像拼图碎片
    draw.ellipse((25, 25, 75, 75), fill=(204, 51, 51, 255))
    draw.ellipse((35, 35, 65, 65), outline=(255, 255, 255, 255), width=3)
    
    return image

# 生成完整拼图图片，包含碎片的位置
def generate_full_puzzle():
    # 创建一个400x400的蓝色图片
    image = Image.new('RGBA', (400, 400), (0, 0, 255, 255))
    draw = ImageDraw.Draw(image)
    
    # 网格图案
    for i in range(9):
        position = i * 50
        # 水平线
        draw.line([(0, position), (400, position)], fill=(229, 229, 229, 128), width=1)
        # 垂直线
        draw.line([(position, 0), (position, 400)], fill=(229, 229, 229, 128), width=1)
    
    # 在右上角画一个红色方块，表示碎片应该在的位置
    draw.rectangle((240, 120, 340, 220), fill=(255, 0, 0, 255))
    
    # 在红色方块内添加与碎片相同的图案
    draw.ellipse((265, 145, 315, 195), fill=(204, 51, 51, 255))
    draw.ellipse((275, 155, 305, 185), outline=(255, 255, 255, 255), width=3)
    
    return image

# 保存图片到文件
def save_image(image, filename):
    current_dir = os.getcwd()
    file_path = os.path.join(current_dir, filename)
    
    image.save(file_path)
    print(f"图片已保存到: {file_path}")

# 生成并保存图片
puzzle_piece = generate_puzzle_piece()
full_puzzle = generate_full_puzzle()

save_image(puzzle_piece, "puzzle_piece.png")
save_image(full_puzzle, "full_puzzle.png")

print("生成测试图片完成") 

File: generate_test_images.swift
import UIKit
import Foundation

// 生成拼图碎片图片
func generatePuzzlePiece() -> UIImage {
    let puzzlePieceSize = CGSize(width: 100, height: 100)
    UIGraphicsBeginImageContextWithOptions(puzzlePieceSize, false, 2.0)
    let puzzleContext = UIGraphicsGetCurrentContext()!
    
    // 背景
    puzzleContext.setFillColor(UIColor.red.cgColor)
    puzzleContext.fill(CGRect(origin: .zero, size: puzzlePieceSize))
    
    // 添加一些细节使其看起来像拼图碎片
    puzzleContext.setFillColor(UIColor(red: 0.8, green: 0.2, blue: 0.2, alpha: 1.0).cgColor)
    puzzleContext.fillEllipse(in: CGRect(x: 25, y: 25, width: 50, height: 50))
    
    puzzleContext.setStrokeColor(UIColor.white.cgColor)
    puzzleContext.setLineWidth(3.0)
    puzzleContext.strokeEllipse(in: CGRect(x: 35, y: 35, width: 30, height: 30))
    
    let puzzlePieceImage = UIGraphicsGetImageFromCurrentImageContext()!
    UIGraphicsEndImageContext()
    
    return puzzlePieceImage
}

// 生成完整拼图图片，包含碎片的位置
func generateFullPuzzle() -> UIImage {
    let fullPuzzleSize = CGSize(width: 400, height: 400)
    UIGraphicsBeginImageContextWithOptions(fullPuzzleSize, false, 2.0)
    let fullContext = UIGraphicsGetCurrentContext()!
    
    // 背景
    fullContext.setFillColor(UIColor.blue.cgColor)
    fullContext.fill(CGRect(origin: .zero, size: fullPuzzleSize))
    
    // 网格图案
    fullContext.setStrokeColor(UIColor(white: 0.9, alpha: 0.5).cgColor)
    fullContext.setLineWidth(1.0)
    
    for i in 0...8 {
        let position = CGFloat(i) * 50.0
        
        // 水平线
        fullContext.move(to: CGPoint(x: 0, y: position))
        fullContext.addLine(to: CGPoint(x: fullPuzzleSize.width, y: position))
        
        // 垂直线
        fullContext.move(to: CGPoint(x: position, y: 0))
        fullContext.addLine(to: CGPoint(x: position, y: fullPuzzleSize.height))
    }
    fullContext.strokePath()
    
    // 在右上角画一个红色方块，表示碎片应该在的位置
    fullContext.setFillColor(UIColor.red.cgColor)
    fullContext.fill(CGRect(x: 240, y: 120, width: 100, height: 100))
    
    // 在红色方块内添加与碎片相同的图案
    fullContext.setFillColor(UIColor(red: 0.8, green: 0.2, blue: 0.2, alpha: 1.0).cgColor)
    fullContext.fillEllipse(in: CGRect(x: 265, y: 145, width: 50, height: 50))
    
    fullContext.setStrokeColor(UIColor.white.cgColor)
    fullContext.setLineWidth(3.0)
    fullContext.strokeEllipse(in: CGRect(x: 275, y: 155, width: 30, height: 30))
    
    let fullPuzzleImage = UIGraphicsGetImageFromCurrentImageContext()!
    UIGraphicsEndImageContext()
    
    return fullPuzzleImage
}

// 保存图片到文件
func saveImage(_ image: UIImage, to fileName: String) {
    guard let data = image.pngData() else {
        print("无法创建PNG数据")
        return
    }
    
    // 获取当前脚本所在目录
    let currentDirectoryURL = URL(fileURLWithPath: FileManager.default.currentDirectoryPath)
    let fileURL = currentDirectoryURL.appendingPathComponent(fileName)
    
    do {
        try data.write(to: fileURL)
        print("图片已保存到: \(fileURL.path)")
    } catch {
        print("保存图片失败: \(error)")
    }
}

// 生成并保存图片
let puzzlePiece = generatePuzzlePiece()
let fullPuzzle = generateFullPuzzle()

saveImage(puzzlePiece, to: "puzzle_piece.png")
saveImage(fullPuzzle, to: "full_puzzle.png")

print("生成测试图片完成") 